includes/class-wc-payments-customer-service.php

```diff
@@ -47,12 +79,16 @@ public function get_customer_id_by_user_id( $user_id ) {
 			return null;
 		}
 
-		$customer_id = get_user_option( self::WCPAY_CUSTOMER_ID_OPTION, $user_id );
+		$customer_id = get_user_option( $this->get_customer_id_option(), $user_id );
+
+		// If customer_id is false it could mean that it hasn't been migrated from the deprecated key.
 		if ( false === $customer_id ) {
-			return null;
+			$customer_id = $this->maybe_migrate_deprecated_customer( $user_id );
```

Comment:
Great change, so well placed intention, love it. But why we re-assigning the variable, does it makes sense?

Reply:
It doesn't. At first, I wanted to return the customer ID from the migration function, but it would be clearer if we just re-fetched it. It's a left over from that, that I didn't catch. Fixed in https://github.com/Automattic/woocommerce-payments/pull/972/commits/5d07c1e7828d734d93f4019cd4f435c6488b059d.